dfastmi/gui/dialog_model.py

```diff
@@ -291,18 +291,17 @@ def _get_condition_configuration(
         measure_files: List,
     ) -> None:
         """Get condition configuration."""
-        num_files = min(len(reference_files), len(measure_files))
-
-        for i, discharge in enumerate(reach.hydro_q[:num_files]):
-            if discharge in reference_files.keys():
+        for i, discharge in enumerate(reach.hydro_q):
+            qstr = str(discharge)
+            if qstr in reference_files.keys() or qstr in measure_files.keys():
```

Comment:
The reference_files and measure_files are handled like they are dictionaries, but the typehint suggest they are lists, which is correct?

Reply:
If it is a dictionary you can also make use of the .get method:
https://www.w3schools.com/python/ref_dictionary_get.asp
You can give it a default, it will return that value if it cannot find the key.
You can remove the two if statements and write it like this:

```
condition = ConditionConfig(
                    Discharge=discharge, 
                    Reference=reference_files.get(qstr, ""), 
                    WithMeasure=measure_files.get(qstr, "")
                )
```

If it cannot find the key it will return "" in this case.

Reply:
These variables may have been lists in the legacy code, but have switched to dictionaries a long time ago. I'm surprised that nothing complains about the mismatch with the type hints. I've used "mypy" in the past for checking type hints. I'm not sure whether that's the best option.

A quick check shows that mypy tool gives 356 warnings ... new issue created: [DFAST-214](https://issuetracker.deltares.nl/browse/DFAST-214) This list/dictionary issue doesn't appear in that list of warning, but it does include a warning concerning
`batch\FileNameRetrieverFactory.py:45: error: Need type annotation for "_creators" (hint: "_creators: dict[<type>, <type>] = ...")`
which may be related.

In dialog_view_model the type hints indicate that it is Dict[float, str] but the key could be an integer (0,1,2) for the legacy mode, a float representing the discharge in the standard mode, and a tuple of a float and a string (discharge and tidal boundary condition) when running in the research mode for the delta reaches of our rivers. Hence, this needs careful crafting.

Reply:
Correction:

1. the mypy listing does include a warning on this topic
`dfastmi\gui\dialog_model.py:287: error: No overload variant of "__getitem__" of "list" matches argument type "str"  [call-overload]`
2. The discharge (float) is converted to str before using as key.

Reply:
Introduced a new type FilenameDict in dfastmi/kernel/typehints.py and used it in DialogModel and DialogViewModel. We'll probably need to use it at other locations as well.

Reply:
Please do not use dictionaries. This breaks the principle of objected programming.

Reply:
Reviewed with Ralph, see #98.

Reply:
Discussed with Bert.